### PR TITLE
Fix Mod checker crash on RekMod

### LINKED ISSUE #10340: NullPointerException in Local mod Errors
**Platform**
Windows, Android

**Version**
4.8.14 (924)

**To Reproduce**
1. I played on 4.8.7 version before, it was okay
2. Downloaded new 4.8.14 (924)
3. Menu -> Options -> Local mod Errors - Crash

<summary>**Platform:** Desktop
**Version:** 4.8.14 (Build 924)
**Rulesets:** [Civ V - Vanilla, RekMOD, Civ V - Gods & Kings]
**Last Screen:** `com.unciv.ui.screens.mainmenuscreen.MainMenuScreen`

--------------------------------

OS: Windows 10
	Windows 10 Enterprise Version 22H2 (Build 19045)
Java: Eclipse Adoptium Temurin-11.0.20+8
	Max Memory: 1024 MB
System default encoding: windows-1251

--------------------------------

**Message:**
```
java.lang.NullPointerException
	at com.unciv.models.ruleset.unique.Unique.conditionalApplies$ruleset(Unique.kt:143)
	at com.unciv.models.ruleset.unique.Unique.conditionalApplies(Unique.kt:180)
	at com.unciv.models.ruleset.unique.Unique.conditionalsApply(Unique.kt:65)
	at com.unciv.models.ruleset.unique.IHasUniques$getMatchingUniques$1.invoke(IHasUniques.kt:34)
	at com.unciv.models.ruleset.unique.IHasUniques$getMatchingUniques$1.invoke(IHasUniques.kt:34)
	at kotlin.sequences.FilteringSequence$iterator$1.calcNext(Sequences.kt:171)
	at kotlin.sequences.FilteringSequence$iterator$1.hasNext(Sequences.kt:194)
	at com.unciv.models.ruleset.validation.RulesetValidator.checkUnitRulesetSpecific(RulesetValidator.kt:663)
	at com.unciv.models.ruleset.validation.RulesetValidator.addUnitErrors(RulesetValidator.kt:456)
	at com.unciv.models.ruleset.validation.RulesetValidator.getErrorList(RulesetValidator.kt:61)
	at com.unciv.models.ruleset.Ruleset.checkModLinks(Ruleset.kt:438)
	at com.unciv.ui.popups.options.ModCheckTab$runModChecker$1.invokeSuspend(ModCheckTab.kt:101)
	at com.unciv.ui.popups.options.ModCheckTab$runModChecker$1.invoke(ModCheckTab.kt)
	at com.unciv.ui.popups.options.ModCheckTab$runModChecker$1.invoke(ModCheckTab.kt)
	at com.unciv.utils.ConcurrencyKt$launchCrashHandling$1.invokeSuspend(Concurrency.kt:87)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
	at com.unciv.utils.CrashHandlingDispatcher$dispatch$1.invoke(Concurrency.kt:173)
	at com.unciv.utils.CrashHandlingDispatcher$dispatch$1.invoke(Concurrency.kt:173)
	at com.unciv.ui.crashhandling.CrashHandlingExtensionsKt$wrapCrashHandling$1.invoke(CrashHandlingExtensions.kt:17)
	at com.unciv.ui.crashhandling.CrashHandlingExtensionsKt$wrapCrashHandlingUnit$1.invoke(CrashHandlingExtensions.kt:33)
	at com.unciv.ui.crashhandling.CrashHandlingExtensionsKt$wrapCrashHandlingUnit$1.invoke(CrashHandlingExtensions.kt:33)
	at com.unciv.utils.CrashHandlingDispatcher.dispatch$lambda$0(Concurrency.kt:173)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)

```
</summary>

```

```
</details>
Platform: Android
Version: 4.8.14 (Build 924)
Rulesets: [Civ V - Vanilla, Civ V - Gods & Kings, RekMOD iron]
Last Screen: com.unciv.ui.screens.mainmenuscreen.MainMenuScreen

--------------------------------

Device Model: Mi 9T
API Level: 29

--------------------------------

**Message:**
```
java.lang.NullPointerException
 at com.unciv.models.ruleset.unique.Unique.conditionalApplies$ruleset(Unique.kt:143)
 at com.unciv.models.ruleset.unique.Unique.conditionalApplies(Unique.kt:180)
 at com.unciv.models.ruleset.unique.Unique.conditionalsApply(Unique.kt:65)
 at com.unciv.models.ruleset.unique.IHasUniques$getMatchingUniques$1.invoke(IHasUniques.kt:34)
 at com.unciv.models.ruleset.unique.IHasUniques$getMatchingUniques$1.invoke(IHasUniques.kt:34)
 at kotlin.sequences.FilteringSequence$iterator$1.calcNext(Sequences.kt:171)
 at kotlin.sequences.FilteringSequence$iterator$1.hasNext(Sequences.kt:194)
 at com.unciv.models.ruleset.validation.RulesetValidator.checkUnitRulesetSpecific(RulesetValidator.kt:663)
 at com.unciv.models.ruleset.validation.RulesetValidator.addUnitErrors(RulesetValidator.kt:456)
 at com.unciv.models.ruleset.validation.RulesetValidator.getErrorList(RulesetValidator.kt:61)
 at com.unciv.models.ruleset.Ruleset.checkModLinks(Ruleset.kt:438)
 at com.unciv.ui.popups.options.ModCheckTab$runModChecker$1.invokeSuspend(ModCheckTab.kt:101)
 at com.unciv.ui.popups.options.ModCheckTab$runModChecker$1.invoke(Unknown Source:8)
 at com.unciv.ui.popups.options.ModCheckTab$runModChecker$1.invoke(Unknown Source:4)
 at com.unciv.utils.ConcurrencyKt$launchCrashHandling$1.invokeSuspend(Concurrency.kt:87)
 at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
 at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
 at com.unciv.utils.CrashHandlingDispatcher$dispatch$1.invoke(Concurrency.kt:173)
 at com.unciv.utils.CrashHandlingDispatcher$dispatch$1.invoke(Concurrency.kt:173)
 at com.unciv.ui.crashhandling.CrashHandlingExtensionsKt$wrapCrashHandling$1.invoke(CrashHandlingExtensions.kt:17)
 at com.unciv.ui.crashhandling.CrashHandlingExtensionsKt$wrapCrashHandlingUnit$1.invoke(CrashHandlingExtensions.kt:33)
 at com.unciv.ui.crashhandling.CrashHandlingExtensionsKt$wrapCrashHandlingUnit$1.invoke(CrashHandlingExtensions.kt:33)
 at com.unciv.utils.CrashHandlingDispatcher.dispatch$lambda$0(Concurrency.kt:173)
 at com.unciv.utils.CrashHandlingDispatcher.$r8$lambda$GFMOlD6QMgmLfgwAvPAW33Ob6HE(Unknown Source:0)
 at com.unciv.utils.CrashHandlingDispatcher$$ExternalSyntheticLambda0.run(Unknown Source:2)
 at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
 at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
 at java.lang.Thread.run(Thread.java:919)
```
Unciv_LPCre7EnmM

### PR DESCRIPTION
**Pull request by SomeTroglodyte**
Closes #10340

### CODE REVIEW

File: core/src/com/unciv/models/ruleset/unique/Unique.kt [hunk 1]
```diff
@@ -161,120 +159,129 @@ class Unique(val text: String, val sourceObjectType: UniqueTarget? = null, val s
             return 0
         }
 
-        return when (condition.type) {
-            // These are 'what to do' and not 'when to do' conditionals
-            UniqueType.ConditionalTimedUnique -> true
+        /** readability shortcuts */
+        fun param0() = condition.params[0]
```
Comment (SeventhM) [inline core/src/com/unciv/models/ruleset/unique/Unique.kt:163]: So I understand the attempt, and I understand we call params 0 and 1 constantly... But is this necessary? This is different from how we call params literally everywhere else

Comment (SeventhM) [reply core/src/com/unciv/models/ruleset/unique/Unique.kt:163]: We just use `unique.params[]`.

Comment (SeventhM) [reply core/src/com/unciv/models/ruleset/unique/Unique.kt:163]: E.g. `UniqueType.BuyUnitsIncreasingCost` uses more than 2 params, seen [here](https://github.com/yairm210/Unciv/blob/8d0853f0a951e33524f3ab5c0662f8b9b1da6431/core/src/com/unciv/logic/city/CityConstructions.kt#L672)

**Comment by SeventhM** [comment]
Besides the subjective discussion of the unnecessary helpers and whether or not that one conditional should work for policies, looks fine to me

File: core/src/com/unciv/models/ruleset/unique/UniqueType.kt [hunk 2]
```diff
@@ -617,6 +617,7 @@ enum class UniqueType(val text: String, vararg targets: UniqueTarget, val flags:
     ConditionalTech("after discovering [tech]", UniqueTarget.Conditional),
     ConditionalNoTech("before discovering [tech]", UniqueTarget.Conditional),
 
+    ConditionalFirstCivToAdopt("if no other Civilization has adopted this", UniqueTarget.Conditional),
```
Comment (yairm210) [inline core/src/com/unciv/models/ruleset/unique/UniqueType.kt:620]: Why are we adding an extra unique as part of this change? Sounds like an entirely separate pr

**Pull request merged**